tools/node-modules

```diff
@@ -0,0 +1,185 @@
+#!/bin/sh
+
+set -eu
+
+CACHE_DIR="${XDG_CACHE_HOME-${HOME}/.cache}/cockpit-dev/cache.git"
+CLONE_URL="https://github.com/cockpit-project/cockpit-node_modules"
+
+cmd_remove() {
+    # if we did this for ourselves the rm is enough, but it might be the case
+    # that someone actually used git-submodule to fetch this, so clean up after
+    # that as well.  NB: deinit nicely recreates the empty directory for us.
+    rm -rf node_modules
+    git submodule deinit node_modules
+    rm -rf "$(git rev-parse --absolute-git-dir)/modules/node_modules"
```

Comment:
If you do normal use of git submodules, git will setup a .git file (in the same way as it does for worktrees) which points to the gitdir for the submodule at that location.

This can also happen as a result of `git submodule absorbgitdirs`.  Here's the docs for that command:

```
absorbgitdirs
   If a git directory of a submodule is inside the
   submodule, move the git directory of the submodule into
   its superproject’s $GIT_DIR/modules path and then connect 
   the git directory and its working directory by setting 
   the core.worktree and adding a .git file pointing to the
   git directory embedded in the superprojects git
   directory.
```

Annoyingly, this isn't cleaned up on deinit.

This is basically just an attempt to make sure we're really squeaky clean.  It's part of the "so clean up after that as well..." mentioned in the comment above.